### PR TITLE
New support for mesh file with MED format

### PR DESCRIPTION
**Pull request by latug0**

### CODE REVIEW

File: src/FiniteElementDiscretization.cxx [hunk 1]
```diff
@@ -80,7 +80,7 @@ namespace mfem_mgis {
         params, FiniteElementDiscretization::NumberOfUniformRefinements, 0);
     if (parallel) {
 #ifdef MFEM_USE_MPI
-      auto smesh = std::make_shared<Mesh<false>>(mesh_file.c_str(), 1, 1);
+      auto smesh = loadMeshSequential(mesh_file.c_str(),1,1);
```
Comment (thelfer) [inline src/FiniteElementDiscretization.cxx:83]: remove the `c_str` call here which create a useless temporary `std::string`

File: src/FiniteElementDiscretization.cxx [hunk 2]
```diff
@@ -91,7 +91,7 @@ namespace mfem_mgis {
 #endif /* MFEM_USE_MPI */
     } else {
       this->sequential_mesh =
-          std::make_shared<Mesh<false>>(mesh_file.c_str(), 1, 1);
```
Comment (thelfer) [inline src/FiniteElementDiscretization.cxx:94]: Here too

File: src/FiniteElementDiscretization.cxx [hunk 3]
```diff
@@ -229,4 +229,22 @@ namespace mfem_mgis {
     return fed.getFiniteElementSpace<false>().GetTrueVSize();
   }  // end of getTrueVSize
 
+  std::shared_ptr<Mesh<false>> loadMeshSequential(const std::string& mesh_name,
+                                                  int generate_edges,
+                                                  int refine,
+                                                  bool fix_orientation) {
+#ifdef MFEM_USE_MED
+    if (std::string("med").compare(getFileExt(mesh_name)) == 0) {
```
Comment (thelfer) [inline src/FiniteElementDiscretization.cxx:237]: `getFileExt(mesh_name)=="med"` is much clearer and efficient, isnt'it ?

**Review by thelfer — COMMENTED** [review]
Hi Guillaume,
Just made a few comments :)
Thomas

File: include/MFEMMGIS/FiniteElementDiscretization.hxx [hunk 4]
```diff
@@ -143,6 +143,28 @@ namespace mfem_mgis {
    */
   MFEM_MGIS_EXPORT size_type getTrueVSize(const FiniteElementDiscretization&);
 
+
+  /*!
+   * \brief Extract the file extension
```
[the remark this answers is not in the document]
Comment (latug0) [reply include/MFEMMGIS/FiniteElementDiscretization.hxx:148]: Right.

File: src/FiniteElementDiscretization.cxx (hunk 1 above)
Comment (latug0) [reply src/FiniteElementDiscretization.cxx:83]: OK done.

File: src/FiniteElementDiscretization.cxx (hunk 2 above)
Comment (latug0) [reply src/FiniteElementDiscretization.cxx:94]: Yes.

File: src/FiniteElementDiscretization.cxx (hunk 3 above)
Comment (latug0) [reply src/FiniteElementDiscretization.cxx]: Done

**Pull request merged**